update servicecatalog cli example docs

eodeyemi14 opened the pull request:
*Issue #, if available:*

*Description of changes:*
This change is to update the AWS Service Catalog CLI example documnetation

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

eodeyemi14 committed:
update servicecatalog cli docs

kdaily commented on awscli/examples/servicecatalog/create-constraint.rst:
The filename is misspelled, missing the s in "constraint". The [style guide](https://aws.github.io/aws-cli/docs_styleguide.html#filename) requires the filename to match the exact command name. Since the command is `create-constraint`, this file should be renamed to `create-constraint.rst`. As-is, the example won't be associated with the command and will never be shown to users.

eodeyemi14 replied:
file is renamed.

kdaily commented on awscli/examples/servicecatalog/create-contraint.rst:
```diff
@@ -0,0 +1,23 @@
+            "Type": "LAUNCH",
+            "Owner": "123456789012",
+            "ProductId": "prod-cfrfxmra3vnu4",
+            "PortfolioId": "port-y3fnkeslpoevg"
```
These values don't match the inputs. Shouldn't it be:


```suggestion
            "ProductId": "prod-cfrfxmraxxxxx",
            "PortfolioId": "port-y3fnkesxxxxx"
```

eodeyemi14 replied:
fixed

kdaily commented on awscli/examples/servicecatalog/delete-provisioned-product-plan.rst:
```diff
@@ -0,0 +1,8 @@
+The following ``delete-provisioned-product-plan`` example deletes a provisioned product plan. ::
+
+    aws servicecatalog delete-provisioned-product-plan \
+        --plan-id "plan-cuxae3z6glaui
```
The double quotes aren't matching, which means this command wouldn't run. Here's a fix:

```suggestion
        --plan-id plan-cuxae3z6glaui
```

eodeyemi14 replied:
fixed

kdaily commented on awscli/examples/servicecatalog/search-products.rst:
```diff
@@ -0,0 +1,38 @@
+**To get information about the products to which the caller has access*
```
Missing a `*` at the end, which would cause bold to not be rendered properly.

```suggestion
**To get information about the products to which the caller has access**
```

kdaily commented on awscli/examples/servicecatalog/search-products.rst:
```diff
@@ -0,0 +1,38 @@
+**To get information about the products to which the caller has access*
+
+The following ``search-products `` example gets all the products the caller has access to. ::
```
Extra space after `search-products`:

```suggestion
The following ``search-products`` example gets all the products the caller has access to. ::
```

kdaily commented on awscli/examples/servicecatalog/terminate-provisioned-product.rst:
```diff
@@ -0,0 +1,28 @@
+
+    aws servicecatalog terminate-provisioned-product  \
+        --provisioned-product-id pp-7z4t3k4hxxxxx 
+{
```
This looks extra, should be removed:

```suggestion
```

kdaily commented on awscli/examples/servicecatalog/create-contraint.rst:
```diff
@@ -0,0 +1,23 @@
+        --portfolio-id port-y3fnkesxxxxx \
+        --product-id prod-cfrfxmraxxxxx \
+        --type LAUNCH \
+        --parameters '{"RoleArn" : "arn:aws:iam::123456789012:role/NewLaunchRole"}'   
```
Trailing whitespace should be removed:

```suggestion
        --parameters '{"RoleArn" : "arn:aws:iam::123456789012:role/NewLaunchRole"}'
```

kdaily commented on awscli/examples/servicecatalog/describe-constraint.rst:
```diff
@@ -0,0 +1,20 @@
+The following ``describe-constraint`` example describes a constraint. ::
+
+    aws servicecatalog describe-constraint  \
+        --id "cons-oqp52evr4bfx2g"   
```
Trailing whitespace should be removed:

```suggestion
        --id "cons-oqp52evr4bfx2g"
```

kdaily commented on awscli/examples/servicecatalog/search-products.rst:
```diff
@@ -0,0 +1,38 @@
+                "SupportUrl": ""
+            }
+        ]
+    } 
```
Trailing whitespace should be removed:

```suggestion
    }
```

kdaily commented on awscli/examples/servicecatalog/terminate-provisioned-product.rst:
```diff
@@ -0,0 +1,28 @@
+The following ``terminate-provisioned-product`` example terminates a provisioned product. ::
+
+    aws servicecatalog terminate-provisioned-product  \
+        --provisioned-product-id pp-7z4t3k4hxxxxx 
```
Trailing whitespace should be removed:

```suggestion
        --provisioned-product-id pp-7z4t3k4hxxxxx
```

kdaily commented on awscli/examples/servicecatalog/create-constraint.rst:
```diff
@@ -0,0 +1,23 @@
+
+The following ``create-constraint`` example creates a constraint. ::
+
+    aws servicecatalog create-constraint  \
```
```suggestion
    aws servicecatalog create-constraint \
```

kdaily commented on awscli/examples/servicecatalog/create-provisioned-product-plan.rst:
```diff
@@ -0,0 +1,20 @@
+
+The following ``create-provisioned-product-plan`` example creates a plan. ::
+
+    aws servicecatalog create-provisioned-product-plan  \
```
```suggestion
    aws servicecatalog create-provisioned-product-plan \
```

kdaily commented on awscli/examples/servicecatalog/delete-constraint.rst:
```diff
@@ -0,0 +1,8 @@
+
+The following ``delete-constraint`` example deletes a constraint. ::
+
+    aws servicecatalog delete-constraint  \
```
```suggestion
    aws servicecatalog delete-constraint \
```

kdaily commented on awscli/examples/servicecatalog/describe-constraint.rst:
```diff
@@ -0,0 +1,20 @@
+
+The following ``describe-constraint`` example describes a constraint. ::
+
+    aws servicecatalog describe-constraint  \
```
```suggestion
    aws servicecatalog describe-constraint \
```

kdaily commented on awscli/examples/servicecatalog/describe-portfolio-share-status.rst:
```diff
@@ -0,0 +1,21 @@
+
+The following ``describe-portfolio-share-status`` example describes portfolio share status. ::
+
+    aws servicecatalog describe-portfolio-share-status  \
```
```suggestion
    aws servicecatalog describe-portfolio-share-status \
```

kdaily commented on awscli/examples/servicecatalog/describe-portfolio-shares.rst:
```diff
@@ -0,0 +1,28 @@
+
+The following ``describe-portfolio-shares`` example describes portfolio shares. ::
+
+    aws servicecatalog describe-portfolio-shares  \
```
```suggestion
    aws servicecatalog describe-portfolio-shares \
```

kdaily commented on awscli/examples/servicecatalog/describe-product.rst:
```diff
@@ -0,0 +1,52 @@
+
+The following ``describe-product`` example describes a product. ::
+
+    aws servicecatalog describe-product  \
```
```suggestion
    aws servicecatalog describe-product \
```

kdaily commented on awscli/examples/servicecatalog/describe-provisioned-product-plan.rst:
```diff
@@ -0,0 +1,25 @@
+
+The following ``describe-provisioned-product-plan`` example describes a provisioned product plan. ::
+
+    aws servicecatalog describe-provisioned-product-plan  \
```
```suggestion
    aws servicecatalog describe-provisioned-product-plan \
```

kdaily commented on awscli/examples/servicecatalog/describe-provisioning-parameters.rst:
```diff
@@ -0,0 +1,63 @@
+
+The following ``describe-provisioning-parameters`` example describes provisioning parameters. ::
+
+    aws servicecatalog describe-provisioning-parameters  \
```
```suggestion
    aws servicecatalog describe-provisioning-parameters \
```

kdaily commented on awscli/examples/servicecatalog/list-launch-paths.rst:
```diff
@@ -0,0 +1,23 @@
+
+The following ``list-launch-paths`` example lists launch paths for a portfolio. ::
+
+    aws servicecatalog list-launch-paths  \
```
```suggestion
    aws servicecatalog list-launch-paths \
```

kdaily commented on awscli/examples/servicecatalog/terminate-provisioned-product.rst:
```diff
@@ -0,0 +1,28 @@
+
+The following ``terminate-provisioned-product`` example terminates a provisioned product. ::
+
+    aws servicecatalog terminate-provisioned-product  \
```
```suggestion
    aws servicecatalog terminate-provisioned-product \
```

kdaily commented on awscli/examples/servicecatalog/update-constraint.rst:
```diff
@@ -0,0 +1,22 @@
+
+The following ``update-constraint`` example updates a constraint. ::
+
+    aws servicecatalog update-constraint  \
```
```suggestion
    aws servicecatalog update-constraint \
```

kdaily commented on awscli/examples/servicecatalog/describe-portfolio-shares.rst:
```diff
@@ -0,0 +1,28 @@
+    {
+        "PortfolioShareDetails": [
+            {
+                "PrincipalId": "012345678901",
```
Account ID doesn't follow the suggested pattern from the style guide:

https://aws.github.io/aws-cli/docs_styleguide.html#sanitizing-sensitive-information

eodeyemi14 replied:
fixed

kdaily commented on awscli/examples/servicecatalog/describe-portfolio-shares.rst:
```diff
@@ -0,0 +1,28 @@
+                "SharePrincipals": false
+            },
+            {
+                "PrincipalId": "098765432109",
```
Account ID doesn't follow the suggested pattern from the style guide:

https://aws.github.io/aws-cli/docs_styleguide.html#sanitizing-sensitive-information

eodeyemi14 replied:
fixed

kdaily commented on awscli/examples/servicecatalog/describe-portfolio-share-status.rst:
```diff
@@ -0,0 +1,21 @@
+        "Status": "COMPLETED",
+        "ShareDetails": {
+            "SuccessfulShares": [
+                "210987654321"
```
Account ID doesn't follow the suggested pattern from the style guide:

https://aws.github.io/aws-cli/docs_styleguide.html#sanitizing-sensitive-information

eodeyemi14 replied:
fixed

kdaily commented on awscli/examples/servicecatalog/create-contraint.rst:
```diff
@@ -0,0 +1,23 @@
+        },
+        "ConstraintParameters": "{\"RoleArn\" : \"arn:aws:iam::123456789012:role/LaunchRole\"}",
+        "Status": "CREATING"
+    }
```
File should contain a trailing newline.


```suggestion
    }

```

eodeyemi14 replied:
fixed

kdaily commented on awscli/examples/servicecatalog/delete-provisioned-product-plan.rst:
```diff
@@ -0,0 +1,8 @@
+    aws servicecatalog delete-provisioned-product-plan \
+        --plan-id "plan-cuxae3z6glaui
+
+This command produces no output.
```
File should contain a trailing newline.
```suggestion
This command produces no output.

```

eodeyemi14 replied:
fixed

kdaily commented on awscli/examples/servicecatalog/describe-constraint.rst:
```diff
@@ -0,0 +1,20 @@
+        },
+        "ConstraintParameters": "{\"RoleArn\" : \"arn:aws:iam::123456789012:role/LaunchRole\"}",
+        "Status": "AVAILABLE"
+    }
```
File should contain a trailing newline.
```suggestion
    }

```

eodeyemi14 replied:
fixed

kdaily commented on awscli/examples/servicecatalog/create-provisioned-product-plan.rst:
```diff
@@ -0,0 +1,20 @@
+        "PlanName": "test-plan",
+        "PlanId": "plan-cuxae3z6glaui",
+        "ProvisionProductId": "pp-7kh7wbc4xxxxx",
+        "ProvisionedProductName": "test-params",
```
Based on the command input `-provisioned-product-name test-pp`, shouldn't this have the same provisioned product name?


```suggestion
        "ProvisionedProductName": "test-pp",
```

eodeyemi14 replied:
fixed

kdaily commented on awscli/examples/servicecatalog/describe-constraint.rst:
```diff
@@ -0,0 +1,20 @@
+
+    {
+        "ConstraintDetail": {
+            "ConstraintId": "cons-7tr6gei4bfx2g",
```
The command uses `--id "cons-oqp52evr4bfx2g"` but the output shows `"ConstraintId": "cons-7tr6gei4bfx2g"`. Should these be the same ID?


```suggestion
            "ConstraintId": "cons-oqp52evr4bfx2g",
```

eodeyemi14 replied:
Yes, the Id is the same as ConstraintId

kdaily reviewed (commented):
Thanks for the submission, and apologies for the delay in review. There are quite a few things that need attention before we can move forward with this. I've left inline comments and suggestions, and there are some that apply to all of the files:

### Each example file is missing a "For more information" link

The [style guide](https://aws.github.io/aws-cli/docs_styleguide.html#for-more-information-link)
requires every example to end with at least one link to relevant documentation. Please add a link at the end of each file; see the style guide for the syntax.The linked topic should be relevant to the specific command — not every file should use the same link.

### Resource IDs are unsanitized or inconsistently sanitized

The [style guide](https://aws.github.io/aws-cli/docs_styleguide.html#sanitizing-sensitive-information) requires sensitive information to be replaced with placeholders. Some IDs use an `xxxxx` pattern in the command but show real-looking IDs in the output (
e.g., `create-contraint.rst` uses `port-y3fnkesxxxxx` in the command but `port-y3fnkeslpoevg` in the output). Others like `cons-7tr6gei4bfx2g`, `plan-cuxae3z6glaui`,
`lpv3-y3fnkeslpoevg`, and `rec-sbwnhzld3b2ku` appear unsanitized throughout. 

Please sanitize all IDs consistently across both command and output sections in each file.

eodeyemi14 committed:
Fix constraint creation example in documentation

Corrected the heading and updated example parameters and output values.

eodeyemi14 committed:
Rename create-contraint.rst to create-constraint.rst

eodeyemi14 committed:
Modify delete-constraint example with new constraint ID

Updated the constraint ID in the delete-constraint example and added a reference to the AWS Service Catalog User Guide.

eodeyemi14 committed:
Fix example IDs in describe-constraint.rst

Updated example IDs in describe-constraint documentation.

eodeyemi14 committed:
Revise list-constraints-for-portfolio example output

Updated example output for list-constraints-for-portfolio command and added reference to AWS documentation.

eodeyemi14 committed:
Change constraint ID in update-constraint example

Updated the constraint ID in the update-constraint example.

eodeyemi14 committed:
Update PlanId in create-provisioned-product-plan.rst

eodeyemi14 committed:
Update example plan ID in delete-provisioned-product-plan

eodeyemi14 committed:
Update example for describe-portfolio-share-status

Updated the portfolio share token in the example and added a reference to the AWS documentation.

eodeyemi14 committed:
Add reference link for portfolio sharing information

Added a reference link to the AWS Service Catalog User Guide for portfolio sharing.

eodeyemi14 committed:
Update product IDs in describe-product.rst

eodeyemi14 committed:
Update example plan ID and Path ID in documentation

eodeyemi14 committed:
Update path-id in describe-provisioning-parameters example

eodeyemi14 committed:
Update Launch Path ID in example documentation

eodeyemi14 committed:
Add reference to AWS Service Catalog User Guide

Added a reference to the AWS Service Catalog User Guide for managing products.

eodeyemi14 committed:
Modify terminate-provisioned-product example output

Updated example output with new RecordId, ProvisioningArtifactId, and PathId.

eodeyemi14 committed:
Update awscli/examples/servicecatalog/search-products.rst

Co-authored-by: Kenneth Daily <kdaily@amazon.com>

eodeyemi14 committed:
Update awscli/examples/servicecatalog/search-products.rst

Co-authored-by: Kenneth Daily <kdaily@amazon.com>

eodeyemi14 committed:
Update ProvisionedProductName in example

eodeyemi14 committed:
Fix formatting in create-provisioned-product-plan example

eodeyemi14 committed:
Fix example by removing quotes from plan ID

Removed unnecessary quotes around the plan ID in the example.

kdaily committed:
Remove trailing whitespace

Co-authored-by: Kenneth Daily <kdaily@amazon.com>

eodeyemi14 committed:
Apply suggestion from @kdaily

Co-authored-by: Kenneth Daily <kdaily@amazon.com>

eodeyemi14 committed:
Apply suggestion from @kdaily

Co-authored-by: Kenneth Daily <kdaily@amazon.com>

eodeyemi14 committed:
Apply suggestion from @kdaily

Co-authored-by: Kenneth Daily <kdaily@amazon.com>

eodeyemi14 committed:
Apply suggestion from @kdaily

Co-authored-by: Kenneth Daily <kdaily@amazon.com>

eodeyemi14 committed:
Update successful share ID in describe-portfolio-share-status

eodeyemi14 committed:
Fix formatting in describe-portfolio-shares example

eodeyemi14 committed:
Update PrincipalId in describe-portfolio-shares.rst

eodeyemi14 committed:
Apply suggestion from @kdaily

Co-authored-by: Kenneth Daily <kdaily@amazon.com>

eodeyemi14 committed:
Apply suggestion from @kdaily

Co-authored-by: Kenneth Daily <kdaily@amazon.com>

eodeyemi14 committed:
Apply suggestion from @kdaily

Co-authored-by: Kenneth Daily <kdaily@amazon.com>

eodeyemi14 committed:
Apply suggestion from @kdaily

Co-authored-by: Kenneth Daily <kdaily@amazon.com>

eodeyemi14 committed:
Apply suggestion from @kdaily

Co-authored-by: Kenneth Daily <kdaily@amazon.com>

eodeyemi14 committed:
Apply suggestion from @kdaily

Co-authored-by: Kenneth Daily <kdaily@amazon.com>

eodeyemi14 committed:
Apply suggestion from @kdaily

Co-authored-by: Kenneth Daily <kdaily@amazon.com>

eodeyemi14 committed:
Apply suggestion from @kdaily

Co-authored-by: Kenneth Daily <kdaily@amazon.com>

eodeyemi14 committed:
Update PrincipalId in describe-portfolio-shares.rst

kdaily commented on awscli/examples/servicecatalog/list-launch-paths.rst:
```diff
@@ -0,0 +1,23 @@
+                "Name": "TestPort"
+            }
+        ]
+    }
```
The [style guide](https://aws.github.io/aws-cli/docs_styleguide.html#for-more-information-link) requires every example to end with at least one link to relevant documentation. Please add a link at the end of each file; see the style guide for the syntax.The linked topic should be relevant to the specific command — not every file should use the same link.

kdaily commented on awscli/examples/servicecatalog/create-provisioned-product-plan.rst:
```diff
@@ -0,0 +1,20 @@
+        "ProvisionProductId": "pp-7kh7wbc4xxxxx",
+        "ProvisionedProductName": "test-pp",
+        "ProvisioningArtifactId": "pa-7wz4cu5cxxxxx"
+    }
```
The [style guide](https://aws.github.io/aws-cli/docs_styleguide.html#for-more-information-link) requires every example to end with at least one link to relevant documentation. Please add a link at the end of each file; see the style guide for the syntax.The linked topic should be relevant to the specific command — not every file should use the same link.

kdaily commented on awscli/examples/servicecatalog/delete-provisioned-product-plan.rst:
```diff
@@ -0,0 +1,8 @@
+    aws servicecatalog delete-provisioned-product-plan \
+        --plan-id plan-cuxae3z6gxxxx
+
+This command produces no output.
```
The [style guide](https://aws.github.io/aws-cli/docs_styleguide.html#for-more-information-link) requires every example to end with at least one link to relevant documentation. Please add a link at the end of each file; see the style guide for the syntax.The linked topic should be relevant to the specific command — not every file should use the same link.

kdaily commented on awscli/examples/servicecatalog/describe-provisioned-product-plan.rst:
```diff
@@ -0,0 +1,25 @@
+            "UpdatedTime": "2025-11-09T18:09:46.524000-06:00",
+            "Tags": []
+        }
+    }
```
The [style guide](https://aws.github.io/aws-cli/docs_styleguide.html#for-more-information-link) requires every example to end with at least one link to relevant documentation. Please add a link at the end of each file; see the style guide for the syntax.The linked topic should be relevant to the specific command — not every file should use the same link.

kdaily commented on awscli/examples/servicecatalog/describe-provisioning-parameters.rst:
```diff
@@ -0,0 +1,63 @@
+            }
+        ],
+        "ProvisioningArtifactOutputKeys": []
+    }
```
The [style guide](https://aws.github.io/aws-cli/docs_styleguide.html#for-more-information-link) requires every example to end with at least one link to relevant documentation. Please add a link at the end of each file; see the style guide for the syntax.The linked topic should be relevant to the specific command — not every file should use the same link.

kdaily reviewed (changes requested):
It's really close! Nice job on getting the IDs sanitized and all the other changes applied. Left a few final comments. Once these are resolved we should be good to port it to v2 and release.

eodeyemi14 committed:
add related documentation

kdaily commented on awscli/examples/servicecatalog/delete-constraint.rst:
```diff
@@ -0,0 +1,10 @@
+The following ``delete-constraint`` example deletes a constraint. ::
+
+    aws servicecatalog delete-constraint \
+        --constraint-id cons-7tr6gei4bxxxx
```
Appropriate parameter name is `--id`, `--constraint-id` is not a valid parameter; this is causing the current test failure.

```suggestion
        --id cons-7tr6gei4bxxxx
```

kdaily committed:
Fix invalid parameter
